compiler/test/dotty/tools/dotc/transform/SpaceEngineTest.scala

```diff
@@ -0,0 +1,25 @@
+package dotty.tools
+package dotc
+package transform
+
+import org.junit.*, Assert.*
+
+import core.*, Constants.*, Contexts.*, Decorators.*, Symbols.*, Types.*
+
+class SpaceEngineTest extends DottyTest:
+  @Test def testAdaptTest(): Unit =
+    given Context = ctx
+    val defn = ctx.definitions
+    import defn._
+    val e = patmat.SpaceEngine()
+
+    val BoxedIntType = BoxedIntClass.typeRef
+    val ConstOneType = ConstantType(Constant(1))
+
+    assertTrue(e.isPrimToBox(IntType, BoxedIntType))
+    assertFalse(e.isPrimToBox(BoxedIntType, IntType))
+    assertTrue(e.isPrimToBox(ConstOneType, BoxedIntType))
```

Comment:
Unit tests are nice but could you add actual pattern matching tests that exercise this?

Reply:
No problem.  The real test case I was using is in #13414, where it emits the spurious warnings once `@unchecked` is removed.  I can revise this and give it its own test cases, independent of the `ordinal` change. 👍🏼

Reply:
This test case I wrote just as a development aid, as I kept getting my "simplify adaptType" wrong, so I wanted to test things more directly.  The true behaviour test for this change is what I was referring to above.